lib/createExplorer.js

```diff
@@ -51,7 +54,7 @@ module.exports = function (options) {
       });
   }
 
-  function searchDirectory(directory) {
+  function searchDirectory(directory, lastSearchDirectory) {
```

Comment:
> the nextDirectory variable WILL have a value and will be DIFFERENT than stopDir, so, it will keep searching on other folders

When we decide to stop searching, though, aren't we comparing `directory` to `stopDir` — not `nextDirectory`? So if the value of `directory` is the home directory, it will *first* search that directory, then check and see if it's the same as `stopDir`, then stop.

I guess the real way to play with the algorithm would be to get strong tests in place first, then make sure they pass.